core/builtin.py

```diff
@@ -513,15 +523,37 @@ def Popd(argv, dir_stack):
 
   try:
     os.chdir(dest_dir)
+    PrintDirStack(dir_stack)
   except OSError as e:
     util.error("popd: %r: %s", dest_dir, os.strerror(e.errno))
     return 1
 
   return 0
 
 
+DIRS_SPEC = _Register('dirs')
+DIRS_SPEC.ShortFlag('-c')
+DIRS_SPEC.ShortFlag('-l')
+DIRS_SPEC.ShortFlag('-p')
+DIRS_SPEC.ShortFlag('-v')
+
 def Dirs(argv, dir_stack):
-  print(dir_stack)
+  arg, i = DIRS_SPEC.Parse(argv)
+  if arg.l:
+    util.warn('*** dirs -l not implemented ***')
+  # Following `bash` behavior for order of operations
+  if arg.c:
+    while len(dir_stack) > 0:
+      dir_stack.pop()
```

Comment:
This loop can be replaced with `del dir_stack[:]`

In Python 3 there is `dir_stack.clear()`, but that's the idiomatic way to do it in Python 2.